diff --git a/tests/unit/parser/_pydantic/test_kinesis.py b/tests/unit/parser/_pydantic/test_kinesis.py
index 9da19ed3e0b..9c3e365c5c5 100644
--- a/tests/unit/parser/_pydantic/test_kinesis.py
+++ b/tests/unit/parser/_pydantic/test_kinesis.py
@@ -105,3 +105,28 @@ class DummyModel(BaseModel): ...
         for record in stream_data.Records:
             record.kinesis.data = DummyModel()
             record.decompress_zlib_record_data_as_json()
+
+
+def test_kinesis_stream_event_with_cloud_watch_logs_data_using_envelope():
+    # GIVEN Kinesis Data Stream event with compressed data
+    # such as CloudWatch Logs
+    raw_event = load_event("kinesisStreamCloudWatchLogsEvent.json")
+
+    # WHEN parsing using KinesisDataStreamEvelope to CloudWatchLogsDecode
+    logs = envelopes.KinesisDataStreamEnvelope().parse(raw_event, CloudWatchLogsDecode)
+
+    # THEN logs should be extracted as CloudWatchLogsDecode objects
+    assert isinstance(logs[0], CloudWatchLogsDecode)
+
+
+def test_kinesis_stream_event_with_cloud_watch_logs_data_fails_using_envelope():
+    # GIVEN Kinesis Data Stream event with corrupted compressed data
+    # such as CloudWatch Logs
+    raw_event = load_event("kinesisStreamCloudWatchLogsEvent.json")
+
+    # WHEN parsing using KinesisDataStreamEvelope to CloudWatchLogsDecode
+    # and the data is corrupted
+    raw_event["Records"][0]["kinesis"]["data"] = "eyJ4eXoiOiAiYWJjIn0KH4sIAK25JWgAA6tWqqisUrJSUEpMSlaq5QIAbdJPfw8AAAA="
+    # THEN a ValueError should be thrown
+    with pytest.raises(ValueError):
+        envelopes.KinesisDataStreamEnvelope().parse(raw_event, CloudWatchLogsDecode)
