diff --git a/tests/commands/new/test_build_gui_context.py b/tests/commands/new/test_build_gui_context.py
index fde3f7ef6..ca753998e 100644
--- a/tests/commands/new/test_build_gui_context.py
+++ b/tests/commands/new/test_build_gui_context.py
@@ -406,8 +406,8 @@ def main():
 """,
         pyproject_table_briefcase_app_extra_content="""
 requires = [
-    "PySide6-Essentials~=6.7",
-    # "PySide6-Addons~=6.7",
+    "PySide6-Essentials~=6.8",
+    # "PySide6-Addons~=6.8",
 ]
 test_requires = [
 {% if cookiecutter.test_framework == "pytest" %}
@@ -417,6 +417,8 @@ def main():
 """,
         pyproject_table_macOS="""\
 universal_build = true
+# As of Pyside 6.8, PySide enforces a macOS 12 minimum on wheels.
+min_os_version = "12.0"
 requires = [
     "std-nslog~=1.0.3",
 ]
